Introduce keyboard scroll flag in Virtual Console

Pac72 opened the pull request:
This is the second try to fix #1443, after a first attempt with pull request #1584 that covered only VCButtons.
The approach on this change is completely different: introduce a flag to choose whether or not to use arrow and PgUp/PgDown keys to scroll Virtual Console.
When the flag is disabled, all the keys are available as key combinations on the controls.

Side changes:

- in assignhotkey.ui the former QLineEdit has been replaced by a QLabel to allow capture of all the keys (the QLineEdit consumes at least backspace, delete, home, end, left and right cursor keys)
- the CTRL+key combinations are completely blocked

Closes #1443

Pac72 commented:
Hi @qfulmina , did you hava a chance to try this change?

mcallegari commented:
If I had to do it, I would have added a toggle button in the Virtual Console toolbar with the existing locker icon (the one used in show manager)
When locked, the VC will not scroll. As easy as that.
I haven't reviewed your PR yet since it is quite large for what it does.
Can you please describe your changes briefly?
Thanks

yestalgia reviewed (commented):
Hell yeah bring back the scroll lock!

Pac72 commented:
@mcallegari, I tried to follow your suggestion, but it does not work here, since the change is relevant in operate mode, when console toolbar is not available.

A lot of time passed since the beginnig of this change, so I think a little recap is needed.

@lightyou opened issue #1443 because he could not use PgUp/PgDown and arrows as keyboard shortcuts.
In the steps to reproduce, he mentioned buttons, so I made a first fix that allowed PgUp/Down/arrows to be used as keyboard shortcuts for buttons (PR #1584, exactly 1 year ago :D ).
He tested the change and pointed out that it did not cover controls other than the buttons.
Hence, I produced the change in this PR, that allows binding the Pg/arrow keys as shortcuts for all controls, but, afraid that other users could rely on the original behaviour (Pg/arrows scrolling VC), I introduced a flag to choose whether to stick with the previous behaviour or switch to the new one (Pg/arrows that can be used as shortcuts).
Two weeks ago @yestalgia proposed the use of Scroll Lock, so I made a new change, allowing, at design time, to override Pg/arrows behaviour with Scroll lock even when the setting is "use Pg/arrows as keyboard shortcuts and not to scroll around".
With this change, the behaviour is:
- design mode: Pg/arrows scroll around
- oerate mode:
  - if setting is "use Pg/arrows to scroll" -> Pg/arrows scroll and do not work as key binding
  - if setting is "do not use Pg/arrows to scroll" -> Pg/arrows can be used as shortcuts in controls, BUT Scroll Lock allows switching to scrolling and back

Changes in current branch:
- AssignHotKey ui used a QLineEdit to capture keyboard shortcut; this control consumes a lot of keys that therefore cannot be used for shortcuts; I replaced it with a simple QLabel that does not consume any key (commits dd5e3c0 and 5413a7e)
- in VirtualConsole, use a new custom VCScrollArea instead of the standard QScrollArea to have control over key events; introduce a setting to control whether to use Pg/arrows to scroll or as shortcuts (commit 1f1409b and e0183cc)
- in operate mode, even when Pg/arrows are configured to be used as shortcuts, allow using them to scroll around pressing the Scroll Lock key (commit 89a98a6)

lightyou commented:
Hello. As the requester for #1443, I can give my opinion. 
Personally I moved away from QLC+ since few months, for several reasons that are not the purpose of this PR, so do not do it only for me. I completely agree that the software shall stay simple enough for common usages, to keep maintenable in the long term.
On other hand, the current behavior is, in my opinion, clearly a bug (as some keys are managed diffently than some others). So this can be either fixed or clearly documented.

Regarding my personal use case, it is to use remote control used for PowerPoint like tools, that send key strokes pgup and pgdown. I have no idea if this use case can become common or not...
